src/state_control/state_control_Lara.cpp

```diff
@@ -3398,39 +3501,204 @@ int StateControl_LaraDoTwoHandWeponFrame(struct entity_s *ent, struct  ss_animat
         float target_pos[3];
         float range = 32768.0f;
         int inc_state;
-        int reload_snd = 0;
-        int ver = World_GetVersion();
-        int num_shots = 1;
-        int shot_snd = 8;
-        int draw_snd = 0;
-        int hide_snd = 0;
-
-        if(ver < TR_II)
-        {
-            if(ss_anim->model->id == 2)
-            {
-                range = 16384.0f;
-                reload_snd = 9;
-                num_shots = 12;
-                draw_snd = 7;
-                hide_snd = 6;
-            }
-        }
-        else
+        int32_t ver = World_GetVersion();
+        int shot_snd = 8, echo_snd = 0, draw_snd = 0, hide_snd = 0, num_shots = 1, reload_snd = 0;
+
+        // check actual version
+        switch (ver)
         {
-            if(ss_anim->model->id == 3)
-            {
-                range = 16384.0f;
-                reload_snd = 9;
-                num_shots = 12;
-            }
+            case TR_I:
```

Comment:
good idea, but it will be better to move this logic to the function like StateControl_LaraGetWeaponSounds(xxx)
more easy to read

Reply:
i will move it to a new function, thanks.
PS: i cant be more clean with switch statement :x (ps: can be possible to add switch in lua :D)

Reply:
you can look the code entirely here:
[state_control_lara.cpp](https://github.com/TokyoSU/OpenTomb/blob/81147416347a3a19772e1d00360c7238acdab286/src/state_control/state_control_Lara.cpp#L3818)